tt_metal/llrt/llrt.hpp

```diff
@@ -98,33 +103,32 @@ void write_circular_buffer_config_vector_to_core(
 
 void program_risc_startup_addr(chip_id_t chip_id, const CoreCoord &core);
 
-bool test_load_write_read_risc_binary(
-    ll_api::memory &mem, chip_id_t chip_id, const CoreCoord &core, int riscv_id);
+bool test_load_write_read_risc_binary(ll_api::memory &mem, chip_id_t chip_id, const CoreCoord &core, int riscv_id);
 
-bool test_load_write_read_trisc_binary(
-    ll_api::memory &mem, chip_id_t chip_id, const CoreCoord &core, int triscv_id);
+bool test_load_write_read_trisc_binary(ll_api::memory &mem, chip_id_t chip_id, const CoreCoord &core, int triscv_id);
 
 // subchannel hard-coded to 0 for now
 CoreCoord get_core_for_dram_channel(int dram_channel_id, chip_id_t chip_id = 0);
 
 namespace internal_ {
 
-void wait_until_cores_done(chip_id_t device_id,
-                           int run_state,
-                           std::unordered_set<CoreCoord>& not_done_phys_cores);
+void wait_until_cores_done(chip_id_t device_id, int run_state, std::unordered_set<CoreCoord> &not_done_phys_cores);
 
 }  // namespace internal_
 
 inline uint64_t relocate_dev_addr(uint64_t addr, uint64_t local_init_addr) {
-
     uint64_t relo_addr;
     if ((addr & MEM_LOCAL_BASE) == MEM_LOCAL_BASE) {
         // Move addresses in the local memory range to l1 (copied by kernel)
         relo_addr = (addr & ~MEM_LOCAL_BASE) + local_init_addr;
-    } else if ((addr & MEM_NCRISC_IRAM_BASE) == MEM_NCRISC_IRAM_BASE) {
+    }
+#ifdef NCRISC_HAS_IRAM
+    else if ((addr & MEM_NCRISC_IRAM_BASE) == MEM_NCRISC_IRAM_BASE) {
```

Comment:
@pgkeller similar to above, I didn't define `MEM_NCRISC_IRAM_BASE` for blackhole which is why I needed the ifdef but I can define it